File: 240328/softeer-6292.cpp
#include<iostream>

using namespace std;

long long number = 1000000007;

long long recur(long long p, long long n){
    if(n == 1){
        return p;
    } 
    long long a;
    a = recur(p, n/2);
    a = (a*a) % number;
    if(n%2 == 1){
        a = (a*p) % number;
    }
    return a;
}

int main(int argc, char** argv)
{
    long long K, P, N;
    cin >> K >> P >> N;

    long long result;
    result = recur(P, 10*N);

    cout << (result*K) % number;
    return 0;
}

File: 240328/softeer-6292.py
import sys
input = sys.stdin.readline

number = 1000000007

def recur(p, n):
    if n == 1:
        return p

    elif n % 2 == 0:
        a = recur(p, n // 2)
        return a * a % number

    else:
        b = recur(p, (n - 1) // 2)
        return b * b * p % number


K, P, N = list(map(int, input().split()))
result = recur(P, 10*N)
result *= K
print(result % number)